code/Model/SiteTree.php

```diff
@@ -2755,6 +2764,21 @@ public function getControllerName() {
 	}
 
 	/**
+	 * Find the frontend controller name, which can be the current fully qualified class name
+	 * with Controller or _Controller added to it within the same namespace, or a customised
+	 * filename as per the $frontend_controller configuration option.
+	 *
+	 * @return string
+	 */
+	public function getFrontendControllerName()
+	{
+		if ($customController = $this->config()->frontend_controller) {
+			return (string) $customController;
+		}
+		return $this->getControllerName();
```

Comment:
This change would need to be made in `SSViewer::get_templates_by_class` as well. In terms of the existing functionality there (look for templates called `MyDataObject_?Controller`) - should we change this to look for a `->getFrontendController()` result, but fall back to the existing method if the class is not a `SiteTree` instance?

It seems to me that if we want to do away with the `_Controller` appending altogether, we should implement this logic higher up than `SiteTree`, probably in `DataObject` instead, then implement the same checks in `SSViewer`